### ISSUE TITLE
Flutter Web VSCode debug

### ISSUE THREAD

**Issue by LucianoMass** [flutter/flutter#35528]
Hi,

I installed the flutter web and when I debug in F5 (VsCode), it creates a random port, launches Chrome, loads the flutter icon in the Chrome bar but the screen goes blank and the "Launching" message stays forever. I left him all night purifying and always stays on the white screen.

If run by the "webdev serve" command, it starts Chrome in the 8080 port and works normally by loading the full page.

Thank you

**Comment by jakemac53** [comment]
If webdev serve worked then this is probably something to do with daemon mode, cc @grouma. I would move it to the webdev repo.

We will need some more information before we can take action though, do you have explicit reproduction steps @LucianoMass ?

**Comment by jakemac53** [comment]
Or it could be the same issue as this one? https://github.com/flutter/flutter/issues/35563#issuecomment-510674256.

I believe the solution here is to downgrade the dart sdk until the next release (it is fixed upstream) but I don't know if it is easy to do that in flutter.

**Comment by LucianoMass** [comment]
yes, it's the same issue...

My steps...

* Open VSCode, Alt+Shift+P,  "Flutter> New Web Project"
* Project name: fw
* Path c:\ (or any folder, I try in 4 folders)
* The project is created.

**OUTPUT**

```
+ analyzer 0.36.4 (0.37.0 available)
+ archive 2.0.10
+ args 1.5.2
+ async 2.2.0
+ bazel_worker 0.1.21
+ build 1.1.4
+ build_config 0.4.0 (0.4.1 available)
+ build_daemon 1.1.0
+ build_modules 2.3.0
+ build_resolvers 1.0.5
+ build_runner 1.6.1
+ build_runner_core 3.0.6
+ build_web_compilers 2.1.2
+ built_collection 4.2.2
+ built_value 6.7.0
+ charcode 1.1.2
+ code_builder 3.2.0
+ collection 1.14.11
+ convert 2.1.1
+ crypto 2.0.6
+ csslib 0.16.1
+ dart_style 1.2.9
+ fixnum 0.10.9
+ front_end 0.1.19 (0.1.20 available)
+ glob 1.1.7
+ graphs 0.2.0
+ html 0.14.0+2
+ http 0.12.0+2
+ http_multi_server 2.1.0
+ http_parser 3.1.3
+ intl 0.15.8
+ io 0.3.3
+ js 0.6.1+1
+ json_annotation 2.4.0
+ kernel 0.3.19 (0.3.20 available)
+ logging 0.11.3+2
+ matcher 0.12.5
+ meta 1.1.7
+ mime 0.9.6+3
+ package_config 1.0.5
+ package_resolver 1.0.10
+ path 1.6.2
+ pedantic 1.8.0+1
+ pool 1.4.0
+ protobuf 0.13.15
+ pub_semver 1.4.2
+ pubspec_parse 0.1.4
+ quiver 2.0.3
+ scratch_space 0.0.3+2
+ shelf 0.7.5
+ shelf_web_socket 0.2.3
+ source_maps 0.10.8
+ source_span 1.5.5
+ stack_trace 1.9.3
+ stream_channel 2.0.0
+ stream_transform 0.0.19
+ string_scanner 1.0.4
+ term_glyph 1.1.0
+ timing 0.1.1+1
+ typed_data 1.1.6
+ vector_math 2.0.8
+ watcher 0.9.7+10
+ web_socket_channel 1.0.14
+ yaml 2.1.16
Warning: You are using these overridden dependencies:
! flutter_web 0.0.0 from git https://github.com/flutter/flutter_web at 8d4c12 in packages/flutter_web
! flutter_web_ui 0.0.0 from git https://github.com/flutter/flutter_web at 8d4c12 in packages/flutter_web_ui
Changed 66 dependencies!
Precompiling executables...
Precompiled build_runner:graph_inspector.
Precompiled build_runner:build_runner.
exit code 

```
**Press F5, select Dart & Flutter (in my vscode have "Node,js" in Environment**

**DEBUG CONSOLE**

```
[INFO] Connecting to the build daemon...
[INFO] Generating build script...
[INFO] Generating build script completed, took 734ms
[INFO]
[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 31.7s
[INFO]
[INFO] Starting daemon...
[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 8.1s
[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 4ms
[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 8.1s
[INFO]
[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 4ms
[INFO]
[INFO] Setting up file watchers...
[INFO] Setting up file watchers completed, took 10ms
[INFO]
[INFO] Registering build targets...
[INFO] Starting initial build...
[INFO] Serving DevTools at http://127.0.0.1:63209
[INFO] Starting resource servers...
[INFO] Serving `web` on http://127.0.0.1:63134
[INFO] About to build [web]...
[INFO] Running build...
[INFO] 1.0s elapsed, 56/56 actions completed.
[INFO] 2.3s elapsed, 76/80 actions completed.
[INFO] 3.5s elapsed, 137/144 actions completed.
[INFO] 4.6s elapsed, 156/176 actions completed.
[INFO] 5.6s elapsed, 226/238 actions completed.
[INFO] 6.7s elapsed, 249/269 actions completed.
[INFO] 7.7s elapsed, 265/285 actions completed.
[INFO] 8.8s elapsed, 273/291 actions completed.
[INFO] 10.2s elapsed, 296/314 actions completed.
[INFO] 11.2s elapsed, 310/328 actions completed.
[INFO] 12.3s elapsed, 343/359 actions completed.
[INFO] 13.4s elapsed, 352/372 actions completed.
[INFO] 14.5s elapsed, 379/396 actions completed.
[INFO] 15.5s elapsed, 410/429 actions completed.
[INFO] 16.6s elapsed, 469/482 actions completed.
[INFO] 17.7s elapsed, 530/545 actions completed.
[INFO] 18.7s elapsed, 555/559 actions completed.
[INFO] 19.7s elapsed, 582/602 actions completed.
[INFO] 20.8s elapsed, 677/679 actions completed.
[INFO] 21.9s elapsed, 695/741 actions completed.
[INFO] 23.9s elapsed, 695/741 actions completed.
[INFO] 25.7s elapsed, 695/741 actions completed.
[INFO] 26.8s elapsed, 695/741 actions completed.
[INFO] 27.9s elapsed, 695/741 actions completed.
[INFO] 29.0s elapsed, 695/741 actions completed.
[INFO] 30.0s elapsed, 698/744 actions completed.
[INFO] 31.0s elapsed, 700/746 actions completed.
[INFO] 32.1s elapsed, 701/746 actions completed.
[INFO] 33.1s elapsed, 703/747 actions completed.
[INFO] 34.3s elapsed, 704/748 actions completed.
[INFO] 35.3s elapsed, 708/750 actions completed.
[INFO] 36.3s elapsed, 709/751 actions completed.
[INFO] 37.3s elapsed, 710/752 actions completed.
[INFO] 38.4s elapsed, 710/752 actions completed.
[INFO] 39.5s elapsed, 710/752 actions completed.
[INFO] 40.5s elapsed, 712/752 actions completed.
[INFO] 41.6s elapsed, 712/752 actions completed.
[INFO] 42.6s elapsed, 712/752 actions completed.
[INFO] 43.6s elapsed, 712/752 actions completed.
[INFO] 44.6s elapsed, 712/752 actions completed.
[INFO] 45.7s elapsed, 712/752 actions completed.
[INFO] 46.7s elapsed, 714/752 actions completed.
[INFO] 47.7s elapsed, 716/752 actions completed.
[INFO] 48.8s elapsed, 717/752 actions completed.
[INFO] 49.9s elapsed, 719/752 actions completed.
[INFO] 50.9s elapsed, 719/752 actions completed.
[INFO] 52.0s elapsed, 719/752 actions completed.
[INFO] 53.1s elapsed, 720/752 actions completed.
[INFO] 54.3s elapsed, 720/752 actions completed.
[INFO] 55.3s elapsed, 721/752 actions completed.
[INFO] 56.3s elapsed, 724/755 actions completed.
[INFO] 57.4s elapsed, 724/755 actions completed.
[INFO] 58.4s elapsed, 725/755 actions completed.
[INFO] 59.5s elapsed, 726/755 actions completed.
[INFO] 1m 0s elapsed, 727/755 actions completed.
[INFO] 1m 1s elapsed, 728/755 actions completed.
[INFO] 1m 2s elapsed, 730/755 actions completed.
[INFO] 1m 3s elapsed, 731/755 actions completed.
[INFO] 1m 4s elapsed, 733/755 actions completed.
[INFO] 1m 5s elapsed, 734/755 actions completed.
[INFO] 1m 7s elapsed, 734/755 actions completed.
[INFO] 1m 8s elapsed, 736/755 actions completed.
[INFO] 1m 9s elapsed, 737/755 actions completed.
[INFO] 1m 10s elapsed, 739/755 actions completed.
[INFO] 1m 11s elapsed, 740/755 actions completed.
[INFO] 1m 12s elapsed, 740/755 actions completed.
[INFO] 1m 13s elapsed, 742/755 actions completed.
[INFO] 1m 14s elapsed, 743/755 actions completed.
[INFO] 1m 15s elapsed, 743/755 actions completed.
[INFO] 1m 16s elapsed, 743/755 actions completed.
[INFO] 1m 17s elapsed, 743/755 actions completed.
[INFO] 1m 18s elapsed, 743/755 actions completed.
[INFO] 1m 19s elapsed, 743/755 actions completed.
[INFO] 1m 20s elapsed, 744/755 actions completed.
[INFO] 1m 21s elapsed, 744/755 actions completed.
[INFO] 1m 22s elapsed, 744/755 actions completed.
[INFO] 1m 23s elapsed, 744/755 actions completed.
[INFO] 1m 24s elapsed, 745/755 actions completed.
[INFO] 1m 26s elapsed, 745/755 actions completed.
[INFO] 1m 27s elapsed, 745/755 actions completed.
[INFO] 1m 28s elapsed, 748/755 actions completed.
[INFO] 1m 29s elapsed, 748/755 actions completed.
[INFO] 1m 30s elapsed, 748/755 actions completed.
[INFO] 1m 31s elapsed, 748/755 actions completed.
[INFO] 1m 32s elapsed, 748/755 actions completed.
[INFO] 1m 33s elapsed, 748/755 actions completed.
[INFO] 1m 34s elapsed, 748/755 actions completed.
[INFO] 1m 35s elapsed, 748/755 actions completed.
[INFO] 1m 36s elapsed, 748/755 actions completed.
[INFO] 1m 37s elapsed, 749/755 actions completed.
[INFO] 1m 38s elapsed, 749/755 actions completed.
[INFO] 1m 40s elapsed, 749/755 actions completed.
[INFO] 1m 41s elapsed, 749/755 actions completed.
[INFO] 1m 42s elapsed, 749/755 actions completed.
[INFO] 1m 43s elapsed, 749/755 actions completed.
[INFO] 1m 44s elapsed, 749/755 actions completed.
[INFO] 1m 45s elapsed, 749/755 actions completed.
[INFO] 1m 46s elapsed, 749/755 actions completed.
[INFO] 1m 47s elapsed, 749/755 actions completed.
[INFO] 1m 48s elapsed, 749/755 actions completed.
[INFO] 1m 49s elapsed, 749/755 actions completed.
[INFO] 1m 50s elapsed, 749/755 actions completed.
[INFO] 1m 51s elapsed, 749/755 actions completed.
[WARNING] No actions completed for 15.0s, waiting on:
  - build_web_compilers:ddc on package:flutter_web/material.ddc.module
  - build_web_compilers:ddc on package:flutter_web/material.ddc.module
  - build_web_compilers:ddc on lib/main.ddc.module
  - build_web_compilers:ddc on web/main.ddc.module
  - build_web_compilers:ddc on lib/main.ddc.module
  .. and 1 more
[INFO] 1m 54s elapsed, 750/755 actions completed.
[INFO] 1m 57s elapsed, 750/755 actions completed.
[INFO] 1m 59s elapsed, 750/755 actions completed.
[INFO] 2m 0s elapsed, 753/755 actions completed.
[INFO] 2m 1s elapsed, 753/755 actions completed.
[INFO] 2m 2s elapsed, 753/755 actions completed.
[INFO] 2m 3s elapsed, 753/755 actions completed.
[INFO] 2m 4s elapsed, 753/755 actions completed.
[INFO] 2m 5s elapsed, 753/755 actions completed.
[INFO] 2m 6s elapsed, 753/755 actions completed.
[INFO] 2m 7s elapsed, 753/755 actions completed.
[INFO] 2m 8s elapsed, 753/755 actions completed.
[INFO] 2m 9s elapsed, 753/755 actions completed.
[INFO] 2m 10s elapsed, 753/755 actions completed.
[INFO] 2m 11s elapsed, 754/755 actions completed.
[INFO] 2m 13s elapsed, 1935/1935 actions completed.
[INFO] 2m 14s elapsed, 2107/2107 actions completed.
[INFO] 2m 15s elapsed, 2244/2244 actions completed.
[INFO] 2m 16s elapsed, 2626/2626 actions completed.
[INFO] 2m 17s elapsed, 2822/2822 actions completed.
[INFO] 2m 18s elapsed, 2978/2978 actions completed.
[INFO] 2m 19s elapsed, 3194/3194 actions completed.
[INFO] Running build completed, took 2m 19s
[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 498ms
[INFO] Succeeded after 2m 20s with 560 outputs (3268 actions)
[INFO] ------------------------------------------------------------------------
```

**The Chrome run**

![image](https://user-images.githubusercontent.com/51490615/61144609-a9676800-a4ab-11e9-87ac-a7d07dcf1f95.png)

 **When I change anything in lib/main.dart, the DEGUG CONSOLE show this**

```
[INFO] About to build [web]...
[INFO] Updating asset graph...
[INFO] Updating asset graph completed, took 13ms
[INFO] Running build...
[INFO] 1.1s elapsed, 1/3 actions completed.
[INFO] Running build completed, took 1.5s
[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 223ms
[INFO] Succeeded after 1.8s with 7 outputs (5 actions)
[INFO] ------------------------------------------------------------------------
Unhandled exception:
FormatException: Unexpected character (at character 1)
Not Found
^
#0      _ChunkedJsonParser.fail  (dart:convert-patch/convert_patch.dart:1392:5)
#1      _ChunkedJsonParser.parseNumber  (dart:convert-patch/convert_patch.dart:1259:9)
#2      _ChunkedJsonParser.parse  (dart:convert-patch/convert_patch.dart:924:22)
#3      _parseJson  (dart:convert-patch/convert_patch.dart:29:10)
#4      JsonDecoder.convert  (dart:convert/json.dart:493:36)
#5      JsonCodec.decode  (dart:convert/json.dart:151:41)
#6      jsonDecode  (dart:convert/json.dart:94:10)
#7      parse
#8      Sources.scriptParsed  (package:dwds/src/sources.dart:55:19)
<asynchronous suspension>
#9      _RootZone.runUnaryGuarded  (dart:async/zone.dart:1314:10)
#10     CastStreamSubscription._onData  (dart:_internal/async_cast.dart:81:11)
#11     _RootZone.runUnaryGuarded  (dart:async/zone.dart:1314:10)
#12     _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:336:11)
#13     _BufferingStreamSubscription._add  (dart:async/stream_impl.dart:263:7)
#14     _SinkTransformerStreamSubscription._add  (dart:async/stream_transformers.dart:68:11)
#15     _EventSinkWrapper.add  (dart:async/stream_transformers.dart:15:11)
#16     WipDomain.eventStream.<anonymous closure>.<anonymous closure>  (package:webkit_inspection_protocol/webkit_inspection_protocol.dart:301:26)
#17     _HandlerEventSink.add  (dart:async/stream_transformers.dart:228:7)
#18     _SinkTransformerStreamSubscription._handleData  (dart:async/stream_transformers.dart:120:24)
#19     _RootZone.runUnaryGuarded  (dart:async/zone.dart:1314:10)
#20     _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:336:11)
#21     _DelayedData.perform  (dart:async/stream_impl.dart:591:14)
#22     _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:707:11)
#23     _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:667:7)
#24     _microtaskLoop  (dart:async/schedule_microtask.dart:41:21)
#25     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:50:5)
#26     _runPendingImmediateCallback  (dart:isolate-patch/isolate_patch.dart:116:13)
#27     _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:173:5)

Exited (255)


```
 **And Chrome in blank screen and degug stops**

**After that, in TERMINAL**

```
PS C:\fw> webdev serve
[INFO] Reading cached asset graph completed, took 569ms
[INFO] Checking for updates since last build completed, took 1.8s
[INFO] Serving `web` on http://127.0.0.1:8080
[INFO] Running build completed, took 592ms
[INFO] Caching finalized dependency graph completed, took 309ms
[INFO] Succeeded after 917ms with 0 outputs (0 actions)
[INFO] -----------------------------------------------------------------------------------------------------------

```
![image](https://user-images.githubusercontent.com/51490615/61144999-c8b2c500-a4ac-11e9-89a7-726509034790.png)

**Comment by grouma** [comment]
Hmm. I think Github swallowed my comment.

This is a duplicate of https://github.com/flutter/flutter/issues/35563.

The root cause was a Windows specific issue that was fixed here: https://github.com/dart-lang/sdk/issues/37364

The next version of the Flutter SDK should fix this. Alternatively you can downgrade your SDK version.
